streaming/consumer.cc

```diff
@@ -65,11 +65,18 @@ reader_consumer_v2 make_streaming_consumer(sstring origin,
                         cf->enable_off_strategy_trigger();
                     }
                     return cf->add_sstable_and_update_cache(sst, offstrategy);
-                }).then([cf, s, sst, use_view_update_path, &vb]() mutable -> future<> {
+                }).then([cf, s, sst, use_view_update_path, &vb, origin] () mutable -> future<> {
                     if (!use_view_update_path) {
                         return make_ready_future<>();
                     }
-                    return vb.local().register_staging_sstable(sst, std::move(cf));
+                    return vb.local().register_staging_sstable(sst, std::move(cf)).then([sst = std::move(sst), origin = std::move(origin)] {
+                        if (origin != sstables::repair_origin) {
+                            return make_ready_future<>();
+                        }
+
+                        // for repair wait until the related view is updated
+                        return sst->wait_until_view_is_built();
```

Comment:
I noticed that while reviewing this change that we may want to consider propagating the `abort_source` more consistently through the streaming path. While this might be outside the scope of the current PR, I wanted to raise it for discussion.

#### Current behavior
- `abort_source` is passed to `stream_manager::init_messaging_service_handler()`
- `mutation_writer::distribute_reader_and_consume_on_shards()` runs in background when handling the `STREAM_MUTATION_FRAGMENTS` verb
- `abort_source` is currently captured and checked in `get_schema_for_write()`
- However, it's not propagated to `streaming::make_streaming_consumer()`

#### Proposal
Consider propagating the `abort_source` from `init_messaging_service_handler()` down to `make_streaming_consumer()` for more consistent abort handling throughout the streaming path.

#### Discussion points
1. Would this additional abort checking be overkill given the existing check in `get_schema_for_write()`?
2. Are there scenarios where we'd want to abort the streaming consumer independently?
3. Would this improve our ability to cleanly shut down streaming operations?

Let me know your thoughts on whether this is worth pursuing in a follow-up PR.

Reply:
This seems like a positive change. But I'm not very sure. @raphaelsc @asias might know better.